src/main/java/com/buglife/sdk/Attachment.java

```diff
@@ -0,0 +1,144 @@
+    public static final String TYPE_PNG = MimeTypes.PNG;
+    public static final String TYPE_JPEG = MimeTypes.JPG;
+    public static final String TYPE_MP4 = MimeTypes.MP4;
+
+    private FileAttachment mFileAttachment;
+
+    private Attachment(@NonNull File file, @NonNull String type) {
+        mFileAttachment = new FileAttachment(file, type);
+    }
+
+    JSONObject getJSONObject() throws JSONException {
+        return mFileAttachment.toJSON();
+    }
+
+    boolean isImageAttachment() {
+        return mFileAttachment.isImage();
+    }
+
+    boolean isVideoAttachment() {
+        return mFileAttachment.isVideo();
+    }
+
+    FileAttachment getFileAttachment() {
+        return mFileAttachment;
+    }
+
+    /* Parcelable */
+
+    @Override public int describeContents() {
+        return 0;
+    }
+
+    @Override public void writeToParcel(Parcel dest, int flags) {
+        dest.writeParcelable(this.mFileAttachment, flags);
+    }
+
+    protected Attachment(Parcel in) {
+        this.mFileAttachment = in.readParcelable(FileAttachment.class.getClassLoader());
+    }
+
+    public static final Creator<Attachment> CREATOR = new Creator<Attachment>() {
+        @Override public Attachment createFromParcel(Parcel source) {
+            return new Attachment(source);
+        }
+
+        @Override public Attachment[] newArray(int size) {
+            return new Attachment[size];
+        }
+    };
+
+    /**
+     * Builder for Attachment objects.
+     */
+    @Deprecated
+    public static class Builder {
+        @NonNull final private String mFilename;
+        @NonNull final private String mType;
+
+        /**
+         * Default constructor.
+         *
+         * @param filename The attachment's filename. This should include the attachment extension
+         * @param type The attachment type
+         */
+        public Builder(@NonNull String filename, @NonNull String type) {
+            mFilename = filename;
+            mType = type;
+        }
+
+        /**
+         * Builds an attachment using a File reference.
+         * @param file The file
+         * @return The attachment
+         */
+        public @NonNull Attachment build(@NonNull File file) {
+            return new Attachment(file, mType);
+        }
+
+        /**
+         * Builds an attachment using a File reference.
+         * @param file The file
+         * @param temporaryFile The file will be deleted when no references to it remain. Calls build(file) if false.
+         * @return The attachment
+         */
+        public @NonNull Attachment build(@NonNull File file, boolean temporaryFile) {
+            return build(file);
+        }
+
+        /**
+         * Builds an image attachment using a bitmap
+         * @param bitmap
+         * @return The attachment
+         */
+        public @NonNull Attachment build(@NonNull Bitmap bitmap) {
+            File file = new File(Buglife.getContext().getCacheDir(), mFilename);
```

Comment:
I hate having to expose an internal API to get a Context but it's the only way to transparently create the file attachment using the old Attachment API 😢